src/igd_async.rs

```diff
@@ -154,8 +161,35 @@ pub fn get_any_address(
                     };
                     // TODO(povilas): make port mapping description configurable
                     gateway
-                        .get_any_address(protocol, socket_addr_v4, 0, "p2p")
-                        .map_err(GetAnyAddressError::RequestPort)
+                        .get_any_address(protocol, socket_addr_v4, lease_duration, "p2p")
+                        .or_else(move |e| {
+                            if let GetAnyAddressError::RequestPort(
+                                AddAnyPortError::OnlyPermanentLeasesSupported
+                            ) = e {
+                                if let Some(timeout) = timeout {
+                                    return {
+                                        gateway
```

Comment:
Would be nice to extract this functionality to separate function, e.g. `forward_port_and_remove_after(timeout)`.